lib/createsend/utils.py

```diff
@@ -101,12 +101,21 @@ def connect(self):
 
         cert_path = os.path.join(os.path.dirname(__file__), 'cacert.pem')
 
-        self.sock = ssl.wrap_socket(
-            sock,
-            self.key_file,
-            self.cert_file,
-            cert_reqs=ssl.CERT_REQUIRED,
-            ca_certs=cert_path)
+        # for >= py3.7, mandatory since 3.12
+        if hasattr(ssl.SSLContext, 'wrap_socket'):
+            context = ssl.SSLContext()
```

Comment:
Hi @riklaunim - thank you for putting this PR forward - we really appreciate it and I can see it is looking good and working with higher versions in my testing so far.

Our apologies this only came to attention recently. We are eager to get this moving forward for the Python community, merging it in and updating the package so that it can be used with higher versions of Python as you've rightly pointed out.

In some testing, I noticed with Python 2.7.18, with this specific line it was causing this issue:

```
    context = ssl.SSLContext()
TypeError: __new__() takes at least 2 arguments (1 given)
```

I was wondering if it would be ok to replace this line with:
```
context = ssl.SSLContext(ssl.PROTOCOL_TLS)
```

I found by doing that I was able to get around the TypeError. I am by no means an experienced Python developer and this may not be the right thing to do. I am very open to your thoughts and feedback about that in a bid to move this forward as quickly as possible. Merging in your fix is a top priority for us at the moment.

Reply:
Python 2 is long gone (same as Python 3.8 and lower) so you may look at dropping the support for it. And yes, it would have to be replaced to
`context = ssl.SSLContext(ssl.PROTOCOL_TLS)`, should be fine.

Reply:
Excellent - thank you for the feedback @riklaunim - we will most likely drop supporting older versions of Python and update the version of the createsend-python package with your changes included.